### PR TITLE
extend session.end event

### PR DESCRIPTION
**Pull request by fspmarshall**
Adds three new fields to the `session.end` event:

```
{
    "server_addr":  "<host>:<port>",
    "session_start": "<timestamp>",
    "session_stop": "<timestamp>",
    # ...
}
```

See #3542

### CODE REVIEW

**Review by awly — COMMENTED** [review]
`session.startExec` also has a `session.start`/`session.end` even pair. Should those also get updated here?

File: lib/srv/sess.go [hunk 1]
```diff
@@ -298,6 +300,9 @@ func (s *SessionRegistry) leaveSession(party *party) error {
 			events.SessionEnhancedRecording: sess.hasEnhancedRecording,
 			events.SessionParticipants:      sess.exportParticipants(),
 			events.SessionServerHostname:    s.srv.GetInfo().GetHostname(),
+			events.SessionServerAddr:        s.srv.GetInfo().GetAddr(),
```
[the remark this answers is not in the document]
Comment (fspmarshall) [reply lib/srv/sess.go:303]: They are both somewhat loosely defined (and both can conceivably be empty).  `Hostname` will be the result of `os.Hostname` by default, and it can be overridden by setting `nodename` in `teleport.yaml`.  `Addr` is a bit more situational, but I think it is usually the SSH listen address by default, but may be overridden by setting `advertise_ip` in `teleport.yaml`.  `Addr` is always empty for servers which are dialed via reverse tunnel, since (from the perspective of the proxy) they don't have a network address.

Basically, this config file:

```yaml
teleport:
    nodename: example.com
    advertise_ip: 1.2.3.4
```

Should result in these event fields:

```
{ "server_addr": "1.2.3.4:3022", "server_hostname": "example.com", ... }
```

(when dealing with a "normal" non-reversetunnel node)

**Comment by fspmarshall** [comment]
> `session.startExec` also has a `session.start`/`session.end` even pair. Should those also get updated here?

@awly I think I did add the new fields to the `session.end` event in `session.startExec`; am I missing something?

**Review by awly — APPROVED** [review]
Ooops, I somehow missed the `startExec` changes.
LGTM
